lib/kubeclient/common.rb

```diff
@@ -253,16 +254,18 @@ def watch_entities(resource_name, options = {})
       uri = @api_endpoint.merge("#{@api_endpoint.path}/#{@api_version}/#{path}")
 
       params = {}
+      format = options.delete(:raw) ? :text : :json
```

Comment:
I guess a delete isn't necessary.  Misread the code into thinking that everything passed into `options` would be passed as option to `WatchStream`, but it looks like there is a filtered list of options that are allowed.

I can change this from a delete to just a `[]` call if you would prefer, just didn't want `:raw` to pollute things in method calls farther down the stack (again, doesn't look like this would have been the case).